### PR TITLE
[crash-handler] Ignore crashed from verifier

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description

Update the commit hash for Move repo and made crash handler skip the error coming from move-bytecode-verifer.

### Test Plan
Had a one-off rust test to make sure the rust thread_local primitive will behave as expected. We've already test the panic behavior in https://github.com/move-language/move/pull/750

### CODE REVIEW

File: aptos-move/aptos-vm/src/move_vm_ext/vm.rs [hunk 1]
```diff
@@ -104,16 +104,18 @@ impl Deref for MoveVmExt {
     }
 }
 
-pub fn verifier_config(treat_friend_as_private: bool) -> VerifierConfig {
+pub fn verifier_config(_treat_friend_as_private: bool) -> VerifierConfig {
```
Comment (davidiw) [inline aptos-move/aptos-vm/src/move_vm_ext/vm.rs:107]: let's delete this in the calls?

**Review by davidiw — APPROVED** [review]

File: crates/crash-handler/src/lib.rs [hunk 2]
```diff
@@ -44,6 +45,11 @@ fn handle_panic(panic_info: &PanicInfo<'_>) {
     // Wait till the logs have been flushed
     aptos_logger::flush();
 
+    // Do not kill the process if the panics happened at move-bytecode-verifier.
+    if state::get_state() == VMState::VERIFIER {
```
Comment (JoshLind) [inline crates/crash-handler/src/lib.rs:49]: Nice! 😄 Question: I'm not sure how the state is managed on the `move-core-types` side, but are there any concerns around race conditions here? e.g., what happens if another part of the code panics while we're in the verification stage? Will this catch the error?

Comment (JoshLind) [reply crates/crash-handler/src/lib.rs:49]: A brief look at https://github.com/move-language/move/commit/e7e42d9ec1a689cc29c9142fcedd092413e3b59e suggests we should be okay? e.g., we're using thread local storage to set this global variable. And, I assume that that the verifier is not going to context switch away during verification (e.g., it doesn't use any async/await), so nothing else should panic during the verification stage?

File: crates/crash-handler/src/lib.rs [hunk 3]
```diff
@@ -44,6 +45,11 @@ fn handle_panic(panic_info: &PanicInfo<'_>) {
     // Wait till the logs have been flushed
     aptos_logger::flush();
 
+    // Do not kill the process if the panics happened at move-bytecode-verifier.
```
Comment (JoshLind) [inline crates/crash-handler/src/lib.rs:48]: Nit: could we add a TODO here that says once `can_unwind` is stable, we should assert it? Just to be sure we aren't ignoring something that cannot unwind? https://github.com/rust-lang/rust/issues/92988

File: aptos-move/aptos-vm/src/move_vm_ext/vm.rs (hunk 1 above)
Comment (runtian-zhou) [reply aptos-move/aptos-vm/src/move_vm_ext/vm.rs:107]: I'll remove this in a separate PR to make the changes minimal here. Guess we will need to cherrypick this into production branch.

File: crates/crash-handler/src/lib.rs (hunk 2 above)
Comment (runtian-zhou) [reply crates/crash-handler/src/lib.rs:49]: Actually this seems a legit concern here? If another thread panics during the verification we should read out the `VERIFIER` state and thus skip the panic?

Comment (runtian-zhou) [reply crates/crash-handler/src/lib.rs:52]: That was a great call! I have a reproducing example here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3cadcce23944a0b7d326779c921d0022. This demonstrates that each thread is usinge its own global variable so no race condition could happen.

**Review by JoshLind — APPROVED** [review]
After syncing with @runtian-zhou offline, this looks like it's good to go 😄

File: crates/crash-handler/src/lib.rs [hunk 4]
```diff
@@ -44,6 +45,14 @@ fn handle_panic(panic_info: &PanicInfo<'_>) {
...
+    // This is safe because the `state::get_state()` uses a thread_local for storing states. Thus other the state can only be mutated to VERIFIER by the thread that's running the bytecode verifier.
```
Comment (wrwg) [inline crates/crash-handler/src/lib.rs:49]: Thus other the --> Thus the

File: crates/crash-handler/src/lib.rs (hunk 2 above)
Comment (wrwg) [reply crates/crash-handler/src/lib.rs:52]: The beauty of thread locals is that there are no race conditions. However, this reminds of another related concern. Was is if the crash happens in a synchronously called function `f` from an external crate via the verifier? We had long discussions about this also with OtterSec. As long as the called code is [UnwindSave](https://doc.rust-lang.org/std/panic/trait.UnwindSafe.html) we should be good. At least we know that the bytecode verifier only calls core Rust framework code which expect to be UnwindSafe. So what we are doing should only be done for certain primitive kind of code.

**Review by wrwg — APPROVED** [review]

**Pull request merged**